File: wordle-web/reference/wordle.py
import json

with open("/home/jeff//wordle_sorted.json") as wordfile:
    wordlist = json.loads(wordfile.read())

bad_letters = "dieustcl"
not_letters = [
    "ar",
    "",
    "a",
    "",
    ""
]

good_letters = {x for pos in not_letters for x in pos}
perfect_letters = ["", "o", "r", "a", ""]


def check_perfect(word):
    for idx, l in enumerate(perfect_letters):
        if l and word[idx] != l:
            return False
    return True


def check_bad_letters(word):
    for idx, l in enumerate(word):
        if l in bad_letters and l != perfect_letters[idx]:
            return False
    return True


def check_not_letters(word):
    for idx, l in enumerate(word):
        if l in not_letters[idx]:
            return False
    return True


printed = 0
for word in wordlist:
    if not check_perfect(word):
        continue
    if not check_bad_letters(word):
        continue
    if not check_not_letters(word):
        continue
    if not all([x in word for x in good_letters]):
        continue

    print(word)
    printed += 1
    if printed > 10:
        break

File: wordle-web/src/wordle.ts

const checkPerfect = (word:string[], perfectLetters:string[]) => {
    for (let index in perfectLetters) {
        const letter = perfectLetters[index]
        if (letter !== '' && word[index] !== letter) return false;
    }
    return true;
}

const checkBadLetters = (word: string[], badLetters: string[], perfectLetters: string[]) => {
    for (let index in word) {
        const letter = word[index];
        if (badLetters.includes(letter) && letter != perfectLetters[index]) return false;
    }
    return true;
}

const checkNotLetters = (word: string[], notLetters: string[][]) => {
    for (let index in word) {
        const letter = word[index];
        if (notLetters[index].includes(letter)) return false;
    }
    return true;
}

const getSuggestions = (badLetters: string[], notLetters: string[][], perfectLetters: string[], wordList: string[]) => {
    const goodLetters = notLetters.join("").split(",").join('').split('')
    const suggestions: string[] = [];
    console.log(badLetters, notLetters, perfectLetters, goodLetters);
    for (const word of wordList) {
        const wordLetters = word.split("");
        if (!checkPerfect(wordLetters, perfectLetters)) continue;
        if (!checkBadLetters(wordLetters, badLetters, perfectLetters)) continue;
        if (!checkNotLetters(wordLetters, notLetters)) continue;
        if (!goodLetters.every((letter: string) => wordLetters.includes(letter))) continue;
        suggestions.push(word);
        if (suggestions.length >= 3) break;
    }
    return suggestions;
}

export default getSuggestions;